lib/pbench/cli/agent/commands/tools/clear.py

```diff
@@ -0,0 +1,193 @@
+"""
+pbench-clear-tools
+
+This script will remove tools that have been registered.  If no options are
+used, then all tools from the "default" tool group are removed.  Specifying
+a tool name and/or remote host will limit the scope of the removal.
+
+
+"""
+import shutil
+import sys
+import pathlib
+
+import click
+
+from pbench.agent.utils import setup_logging
+from pbench.cli.agent import CliContext, pass_cli_context
+from pbench.cli.agent.commands.tools.base import ToolCommand
+from pbench.cli.agent.options import common_options
+
+
+class ClearTools(ToolCommand):
+    """Clear registered tools"""
+
+    def __init__(self, context):
+        super().__init__(context)
+
+        self.logger = setup_logging(logfile=self.pbench_log)
+
+    def execute(self):
+        errors = 0
+
+        if self.verify_tool_group(self.context.group) != 0:
+            return 1
+
+        try:
+            remotes = self.context.remote.split(",")
+        except Exception:
+            # We were not given any remotes on the command line, build the list
+            # from the tools group directory.
+            remotes = self.remote(self.tool_group_dir)
+
+        for remote in remotes:
+            tg_dir_r = self.tool_group_dir / remote
+            if not tg_dir_r.exists():
+                self.logger.warn(
+                    'The given remote host, "%s", is not a directory in' " %s.",
+                    remote,
+                    self.tool_group_dir,
+                )
+                continue
+
+            if self.context.name:
+                names = self.context.name
+            else:
+                # Discover all the tools registered for this remote
+                names = self.tools(tg_dir_r)
+
+            for name in names:
+                status = self._clear_tools(name, remote)
+                if status != 0:
+                    errors += 1
+
+            tool_files = [p.name for p in tg_dir_r.iterdir()]
+
+            if len(tool_files) == 1 and tool_files[0] == "__label__":
+                label = tg_dir_r / "__label__"
+                try:
+                    label.unlink()
+                except Exception:
+                    self.logger.error("Failed to remove label for remote %s", tg_dir_r)
+                    errors += 1
+
+            if self.is_empty(tg_dir_r):
+                self.logger.info('All tools removed from host, "%s"', tg_dir_r.name)
+                try:
+                    shutil.rmtree(tg_dir_r)
+                except OSError:
+                    self.logger.error("Failed to remove remote directory %s", tg_dir_r)
+                    errors += 1
+
+        return errors
+
+    def _clear_tools(self, name, remote):
+        """Remove specified tool and associated files"""
+        tpath = self.tool_group_dir / remote / name
+        try:
+            if not tpath.exists():
```

Comment:
Not sure we need to place this under the `try` block.  Seems like you could either: a. move it before the `try` block, or b. change the `try` block:

```
try:
    tpath.unlink()
except FileNotFoundError:
    self.logger.debug("Tool \"%s\" not registered for remote \"%s\"", name, remote)
    return 0
except Exception as exc:
    .
    .
    .
```